pyirf/interpolation/moment_morph_interpolator.py

```diff
@@ -0,0 +1,442 @@
+        )
+        w2 = ((-d13[1] * dp3[0]) + (d13[0] * dp3[1])) / (
+            (d23[1] * d13[0]) + (-d23[0] * d13[1])
+        )
+
+        # Use w1+w2+w3 = 1 for third weight
+        w3 = 1 - w1 - w2
+
+        coefficients = np.array([w1, w2, w3])
+
+        return coefficients
+
+
+class Base1DMomentMorphInterpolator(BaseMomentMorphInterpolator):
+    def __init__(self, grid_points, bin_edges, bin_contents):
+        """
+        Base class implementing the MomentMorphInterpolation between two points.
+
+        Parameters
+        ----------
+        grid_points: np.ndarray, shape=(2)
+            Grid points at which interpolation templates exist
+        bin_edges: np.ndarray, shape=(M+1)
+            Edges of the data binning
+        bin_content: np.ndarray, shape=(2, ..., M)
+            Content of each bin in bin_edges for
+            each point in grid_points. First dimesion has to correspond to number
+            of grid_points, last dimension has to correspond to number of bins for
+            the quantity that should be interpolated (e.g. the Migra axis for EDisp)
+
+        Note
+        ----
+            Also calls BaseMomentMorphInterpolator's __init__.
+        """
+        if grid_points.shape != (2, 1):
+            raise ValueError("This base class can only interpolate between two points.")
+
+        super().__init__(grid_points, bin_edges, bin_contents)
+
+    def _target_in_grid(self, target_point):
+        """
+        Override target_in_grid-check of parent class so this BaseClass can natively
+        be used to extrapolate
+        """
+        return True
+
+    def get_interpolation_coefficients(self, target_point):
+        """
+        Compute 1D interpolation coefficients for moment morph interpolation,
+        as in eq. (6) of [1]
+
+        Parameters
+        ----------
+        target_point: numpy.ndarray, shape=(1, 1)
+            Value at which the histogram should be interpolated
+
+        Returns
+        -------
+        coefficients: numpy.ndarray, shape=(2,)
+            Interpolation coefficients for all three interpolation simplex vertices
+            to interpolate to the target_point
+
+        References
+        ----------
+        .. [1] M. Baak, S. Gadatsch, R. Harrington and W. Verkerke (2015). Interpolation between
+               multi-dimensional histograms using a new non-linear moment morphing method
+               Nucl. Instrum. Methods Phys. Res. A 771, 39-48. https://doi.org/10.1016/j.nima.2014.10.033
+        """
+        # Set zeroth grid point as reference value
+        m0 = self.grid_points[0, :]
+
+        # Compute matrix M as in eq. (2) of [1]
+        j = np.arange(0, self.n_points, 1)
+        m_ij = (self.grid_points - m0) ** j
+
+        # Compute coefficients, eq. (6) from [1]
+        return np.einsum(
+            "...j, ji -> ...i", ((target_point - m0) ** j), np.linalg.inv(m_ij)
+        )
+
+
+class MomentMorphInterpolator(BinnedInterpolator):
```

Comment:
I think it's confusing that `MomentMorphInterpolator` is not a child class of `BaseMomentMorph` interpolator. The class hierarchy is a bit unintuitive.